src/filters/price_filter.rs

```diff
@@ -0,0 +1,79 @@
+use super::*;
+
+pub struct PriceFilter {
+    pub locale: String,
+}
+
+impl PriceFilter {
+    fn currencies(&self) -> &'static str {
+        "$¢£¤¥֏؋৲৳৻૱௹฿₠₡₢₣₤₥₦₧₨₩₪₫€₭₮₯₰₱₲₳₴₵₶₷₸₹₺₽﹩＄￠￡￥￦"
+    }
+}
+
+impl LinterFilter for PriceFilter {
+    fn locales(&self) -> Vec<&'static str> {
+        vec!("fr")
+    }
+
+    fn message(&self) -> &'static str {
+        match &self.locale as &str {
+            "fr" => "The currency sign should be written after the amount and a non-breaking space.",
+            _ => "",
+        }
+    }
+
+    fn regex_pattern(&self) -> String {
+        match &self.locale as &str {
+            "fr" => format!("([\\d]+[^ ][{}]{{1}}|[{}]{{1}}[\\s]?[\\d]+)", self.currencies(), self.currencies()),
```

Comment:
Not really; in my opinion, the provided tests help to understand the meaning of the filter, but not the "final" content of this regex. We could create tests for this individual function (that tests a fragment of it results, for exemple?) to help us understand what this regex looks like.

This is mainly the problem of complex regex: they are rarely ligitible. An English-written explanation could also helps a lot to dive into it, like `// Checks for digits followed by non-spaces, then by any of currencies() return value`… would be enough imho.